testsuite/tests/array-functions/test.ml

```diff
@@ -0,0 +1,239 @@
+  assert (does_raise3 Array.exists2 (fun a b -> a = 6 && b = 6) a b);
+  assert (does_raise3 Array.exists2 (fun a b -> a = 7 && b = 7) a b);
+  assert (does_raise3 Array.exists2 (fun a b -> a = 8 && b = 8) a b);
+  assert (does_raise3 Array.exists2 (fun a b -> a = 9 && b = 9) a b);
+;;
+
+let () =
+  let a = [|1;2;3;4;5;6;7;8;9|]  
+  and b = [|1;2;3;4;5;6;7;8;9|] in
+  assert (Array.for_all2 (fun a b -> a = b) a b);
+  assert (Array.for_all2 (fun a b -> a - b = 0) a b);
+  assert (Array.for_all2 (fun a b -> a > 0 && b > 0) a b);
+  assert (Array.for_all2 (fun a b -> a < 10 && b < 10) a b);
+  assert (Array.for_all2 (fun a b -> if a = 1 then b = 1 else b <> 1) a b);
+  assert (Array.for_all2 (fun a b -> if a = 2 then b = 2 else b <> 2) a b);
+  assert (Array.for_all2 (fun a b -> if a = 3 then b = 3 else b <> 3) a b);
+  assert (Array.for_all2 (fun a b -> if a = 4 then b = 4 else b <> 4) a b);
+  assert (Array.for_all2 (fun a b -> if a = 5 then b = 5 else b <> 5) a b);
+  assert (Array.for_all2 (fun a b -> if a = 6 then b = 6 else b <> 6) a b);
+  assert (Array.for_all2 (fun a b -> if a = 7 then b = 7 else b <> 7) a b);
+  assert (Array.for_all2 (fun a b -> if a = 8 then b = 8 else b <> 8) a b);
+  assert (Array.for_all2 (fun a b -> if a = 9 then b = 9 else b <> 9) a b);
+  assert (not (Array.for_all2 (fun a b -> a <> b) a b));
+;;
+
+let () =
+  let a = [|1|]
+  and b = [|1;2|] in
+  assert (does_raise3 Array.for_all2 (fun a b -> a = b) a b);
+  assert (does_raise3 Array.for_all2 (fun _ _ -> true) a b);
+  assert (does_raise3 Array.for_all2 (fun _ _ -> false) a b);
+  assert (does_raise3 Array.for_all2 (fun a b -> a = 1 && b = 1) a b);
+  assert (does_raise3 Array.for_all2 (fun a b -> a = 2 && b = 2) a b);
+  assert (does_raise3 Array.for_all2 (fun a b -> a = 3 && b = 3) a b);
+  assert (does_raise3 Array.for_all2 (fun a b -> a = 4 && b = 4) a b);
+  assert (does_raise3 Array.for_all2 (fun a b -> a = 5 && b = 5) a b);
+  assert (does_raise3 Array.for_all2 (fun a b -> a = 6 && b = 6) a b);
+  assert (does_raise3 Array.for_all2 (fun a b -> a = 7 && b = 7) a b);
+  assert (does_raise3 Array.for_all2 (fun a b -> a = 8 && b = 8) a b);
+  assert (does_raise3 Array.for_all2 (fun a b -> a = 9 && b = 9) a b);
+;;
+
+let () =
+  let a = [|1;2;3;4;5;6;7;8;9|] in
+  assert (Array.mem 1 a);
+  assert (Array.mem 2 a);
+  assert (Array.mem 3 a);
+  assert (Array.mem 4 a);
+  assert (Array.mem 5 a);
+  assert (Array.mem 6 a);
+  assert (Array.mem 7 a);
+  assert (Array.mem 8 a);
+  assert (Array.mem 9 a);
+  assert (not (Array.mem 0 a));
+  assert (not (Array.mem 10 a));
+;;
+
+let () =
+  let a = [|1;2;3;4;5;6;7;8;9|] in
+  assert (Array.memq 1 a);
+  assert (Array.memq 2 a);
+  assert (Array.memq 3 a);
+  assert (Array.memq 4 a);
+  assert (Array.memq 5 a);
+  assert (Array.memq 6 a);
+  assert (Array.memq 7 a);
+  assert (Array.memq 8 a);
+  assert (Array.memq 9 a);
+  assert (not (Array.memq 0 a));
+  assert (not (Array.memq 10 a));
```

Comment:
`mem` and `memq` need some testing on non-immediate values to exercise non-trivial identity. See the [Batteries tests](https://github.com/ocaml-batteries-team/batteries-included/blob/master/src/batArray.mlv#L142-L161), for example.